kombu/transport/redis_cluster.py

```diff
@@ -189,15 +189,23 @@ def _unregister(self, channel, client, conn, cmd):
         sock = self._chan_to_sock[(channel, client, conn, cmd)]
         fd = self._sock_to_fd[sock]
 
+        self.poller.unregister(sock)
         if conn.client:
+            if conn.client.connection:
+                # There might be pending BRPOP response on the connection, so we disconnect to ensure safety
+                conn.client.connection.disconnect()
             conn.client.close()
             conn.client = None
 
         del self._fd_to_chan[fd]
         del self._chan_to_sock[(channel, client, conn, cmd)]
         del self._sock_to_fd[sock]
 
-        self.poller.unregister(sock)
+    def discard(self, channel):
+        super().discard(channel)
+        conns_to_unregister = [conn for conn in self._chan_to_sock if conn[0] == channel]
+        for conn in conns_to_unregister:
+            self._unregister(*conn)
```

Comment:
what is it for? Can you add more description about your change? and it would be great if you add tests for the changes

Reply:
Actually it's added to pass tests. i'll add comments